src/client/cli/cmd/exec.cpp

```diff
@@ -32,26 +53,82 @@ mp::ReturnCode cmd::Exec::run(mp::ArgParser* parser)
         return parser->returnCodeFrom(ret);
     }
 
+    std::string instance_name = ssh_info_request.instance_name(0);
+
     std::vector<std::string> args;
     for (int i = 1; i < parser->positionalArguments().size(); ++i)
         args.push_back(parser->positionalArguments().at(i).toStdString());
 
-    auto on_success = [this, &args](mp::SSHInfoReply& reply) { return exec_success(reply, args, term); };
+    std::optional<std::string> work_dir;
```

Comment:
For now, I think this should be `mp::optional`, but we should change all `multipass::optional` to `std::optional` in another PR. I think the oldest version of macOS we support now has proper `optional` support, so I think we need to bring our code up to speed and get rid of `multipass::optional` and just use `std::optional` everywhere.  I'll open a bug about that so we don't forget :wink: